internal/telemetry/collector.go

```diff
@@ -110,26 +97,47 @@ func (c *Collector) Collect(ctx context.Context) {
 	if err != nil {
 		glog.Errorf("Error exporting telemetry data: %v", err)
 	}
-	glog.V(3).Infof("Exported telemetry data: %x", traceData)
+	glog.V(3).Infof("Exported telemetry data: %+v", traceData)
 }
 
-// Start starts running NIC Telemetry Collector.
-func (c *Collector) Start(ctx context.Context) {
-	wait.JitterUntilWithContext(ctx, c.Collect, c.Period, 0.1, true)
+// BuildReport takes context and builds report from gathered telemetry data.
+func (c *Collector) BuildReport(ctx context.Context) (TraceData, error) {
+	dt := TraceData{}
+	var err error
+
+	for _, namespace := range c.Config.Namespaces {
+		dt.VirtualServers += c.GetVirtualServerCount(ctx, namespace)
+		dt.TransportServers += c.GetTransportServerCount(ctx, namespace)
+	}
+
+	return dt, err
```

Comment:
are we planning to handle errors here? atm the err value it's always `nil`. may be worth to simplify the func signature, unless the err would be used in the future.